### ISSUE TITLE
Technical university of Delft fork

### ISSUE THREAD

**Issue by Era-Dorta** [alan-turing-institute/python-project-template#52]
Hi there,

First, thanks a lot for creating this template, it is really useful.

Prompted by a colleague that used to work there, we have created a fork for the faculty of Electrical Engineering, Mathematics and Computer Science. You can find it here https://gitlab.ewi.tudelft.nl/reit/python-package-template

We've made all the changes to fit our use cases as best as possible.  We also took some inspiration from the NLeSC template https://github.com/NLeSC/python-template

Would you be interested in upstreaming any of the changes that we have done in our fork? We'd be happy to make some PRs.

**Comment by phinate** [comment]
Wow, this is fantastic! I appreciate this a lot from both an endorsement and a contribution perspective :)

I'd love to help work through PR-ing some of this over. I was already planning to do a `uv` re-write, so I'm definitely after that. I've also heard many good things about `pixi` -- what was the use case that made you want to support both?

In a way, `uv` is also not truly a backend yet, as it doesn't have it's own build engine. It's still `hatchling`/`setuptools` doing the legwork. But I'm also keen to support it in a simpler way than just saying "this also works with `uv`", which is sort of what I do now.

TL;DR is: yes!! If we can scope this to maybe the following categories:
- `uv` support
- `pixi` support
- docs
- any other quality-of life stuff you've added

and that would massively help me filter through things.

Thank you for using and improving the template! <3

**Comment by Era-Dorta** [comment]
Amazing! I'll make some PRs but please expect a very slow time frame. I have less than a day a week allocated to work on this and other internal tasks.

> I've also heard many good things about pixi -- what was the use case that made you want to support both?

`uv` is much faster than `pixi` (for solving and installing dependencies) and supports building and pushing wheels to Pypi. So we decided to encourage people to use `uv` if they don't have any conda dependencies in their projects. However, conda is incredibly popular so we wanted to have support for an environment manager that has lock files for it. With the addition that the `rattler` solver that `pixi` uses is much faster than using `mamba`, `miniconda` and similar. The other downside is that `pixi` doesn't support building conda packages yet, as explained here https://pixi.sh/latest/FAQ/#where-is-pixi-build
